syntax = "proto3";

package opencv_object_tracking;

service ObjectLocationServer {

    rpc registerClient (ClientInfo) returns (ServerInfo) {
    }

    rpc getObjectLocations (ClientInfo) returns (stream ObjectLocation) {
    }

}

service FocusLinePositionServer {

    rpc registerClient (ClientInfo) returns (ServerInfo) {
    }

    rpc getFocusLinePositions (ClientInfo) returns (stream FocusLinePosition) {
    }

}

message ClientInfo {
    string info = 1;
}

message ServerInfo {
    string info = 1;
}

message ObjectLocation {
    int32 x = 1;
    int32 y = 2;
    int32 width = 3;
    int32 height = 4;
    int32 middle_inc = 5;
    int32 id = 7;
}

message FocusLinePosition {
    bool in_focus = 1;
    int32 mid_offset = 2;
    int32 degrees = 3;
    int32 mid_line_cross = 4;
    int32 width = 5;
    int32 middle_inc = 6;
}


